core/templates/hash_set.h

```diff
@@ -144,15 +144,11 @@ class HashSet {
 		uint32_t *old_hashes = hashes;
 		uint32_t *old_key_to_hash = key_to_hash;
 
-		hashes = reinterpret_cast<uint32_t *>(Memory::alloc_static(sizeof(uint32_t) * capacity));
+		hashes = reinterpret_cast<uint32_t *>(Memory::alloc_static_zeroed(sizeof(uint32_t) * capacity));
 		keys = reinterpret_cast<TKey *>(Memory::realloc_static(keys, sizeof(TKey) * capacity));
 		key_to_hash = reinterpret_cast<uint32_t *>(Memory::alloc_static(sizeof(uint32_t) * capacity));
 		hash_to_key = reinterpret_cast<uint32_t *>(Memory::realloc_static(hash_to_key, sizeof(uint32_t) * capacity));
 
-		for (uint32_t i = 0; i < capacity; i++) {
-			hashes[i] = EMPTY_HASH;
-		}
-
```

Comment:
Should we `static_assert()` here that `EMPTY_HASH` is zero?

Reply:
Funny, now that you're saying it I remember I was asking the same of another author recently:
https://github.com/godotengine/godot/pull/90082#discussion_r2018651137

So yea, I'll add the appropriate `static_asserts` on the next rebase 😄